### PR TITLE
signer utility

### PR DESCRIPTION
**Pull request by cendhu**
This commit adds an utility that can help the user to add
their digital signature on any data item. This would be used
when user submits query or transaction through `curl` command

Signed-off-by: senthil <cendhu@gmail.com>

### CODE REVIEW

File: pkg/cryptoservice/signer.go [hunk 1]
```diff
@@ -48,14 +49,15 @@ func SignTx(txSigner crypto.Signer, tx interface{}) ([]byte, error) {
...
+	return SignPayload(txSigner, tx)
 }
 
-func signPayload(signer crypto.Signer, payload interface{}) ([]byte, error) {
+func SignPayload(signer crypto.Signer, payload interface{}) ([]byte, error) {
 	payloadBytes, err := json.Marshal(payload)
 	if err != nil {
 		return nil, err
 	}
+	fmt.Println(string(payloadBytes))
```
Comment (C0rWin) [inline pkg/cryptoservice/signer.go:60]: is it a leftover?

**Comment by C0rWin** [comment]
Why can't you use openssl?

```
openssl dgst -ecdsa-with-SHA1 -sign your_private_key.pem data.txt > signature.out
```

**Comment by cendhu** [comment]
Didn't strike me @C0rWin Thanks for pointing this out.

What would be the exact command?

`echo '{\"user_id\":\"admin\",\"db_name\":\"bdb\"}' | openssl dgst -sha256 -sign deployment/sample/crypto/admin/admin.key | openssl base64`

``` 
curl \
     -H "Content-Type: application/json" \
     -H "UserID: admin" \
     -H "Signature: MEUCIQD0qPqQMX8UvPqSL8zZyjEORJIXuMn4zn7zKZ1Dcr+wuwIgIW225rke+YJAtS+Ryc5CrnJq070M4IvvL7jzgbCpNiM=" \
     -X GET http://127.0.0.1:6001/db/bdb
```
returns
```
{
  "error": "signature verification failed"
}
```

If we find out the correct command parameter for the `OpenSSL`, it would be useful to reuse the existing tooling itself.

@gennadylaventman @tock-ibm Can you help with the `OpenSSL` command?

File: cmd/signer/signer.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"log"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/crypto"
+)
+
+var help = "all the following three flags must be set. An example command is shown below: \n\n" +
+	"  signer -data=\"{\\\"userID\\\":\\\"admin\\\"}\" -privatekey=admin.key\n"
```
Comment (tock-ibm) [inline cmd/signer/signer.go:13]: Why 3? I see only 2.

File: cmd/signer/signer_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	"path/filepath"
+	"testing"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/crypto"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/server/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSigner(t *testing.T) {
+	t.Run("incorrect orgs", func(t *testing.T) {
```
Comment (tock-ibm) [inline cmd/signer/signer_test.go:20]: `orgs` -> `args`?

File: cmd/signer/signer_test.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+			require.NoError(t, err)
+
+			expectedOut := help + "\n  -data string\n    \tjson data to be signed. All double quotes must be escaped." +
+				" An example json data is \"{\\\"userID\\\":\\\"admin\\\"}\"\n  -privatekey string\n    \tpath to the private key" +
+				" to be used for adding a digital signature\n"
+			require.Equal(t, expectedOut, string(out))
+		}
+	})
+
+	t.Run("correct org", func(t *testing.T) {
```
Comment (tock-ibm) [inline cmd/signer/signer_test.go:36]: `org` -> `args` ?

File: cmd/signer/signer_test.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+		b, err := ioutil.ReadFile(adminCert)
+		require.NoError(t, err)
+		bl, _ := pem.Decode(b)
+		require.NotNil(t, bl)
+		verifier, err := crypto.NewVerifier(bl.Bytes)
+		require.NoError(t, err)
+
+		signature, err := base64.StdEncoding.DecodeString(string(out))
+		require.NoError(t, err)
+		require.NoError(t, verifier.Verify([]byte("{\"user_id\":\"admin\"}"), signature))
```
Comment (tock-ibm) [inline cmd/signer/signer_test.go:65]:
```
adminCert, adminKey := createAdminCreds(t, tempDir)
		data := "\"{\"user_id\":\"admin\"}\""
		args := []string{
			"run",
			"signer.go",
			"-data=" + data,
			"-privatekey=" + adminKey,
		}

		out, err := exec.Command("go", args...).Output()
		require.NoError(t, err)

		b, err := ioutil.ReadFile(adminCert)
		require.NoError(t, err)
		bl, _ := pem.Decode(b)
		require.NotNil(t, bl)
		verifier, err := crypto.NewVerifier(bl.Bytes)
		require.NoError(t, err)

		signature, err := base64.StdEncoding.DecodeString(string(out))
		require.NoError(t, err)
		require.NoError(t, verifier.Verify([]byte(data), signature))
```
double quotes are not the same in sign and verify...

File: pkg/cryptoservice/signer.go (hunk 1 above)
Comment (cendhu) [reply pkg/cryptoservice/signer.go]: Done.

File: cmd/signer/signer.go (hunk 2 above)
Comment (cendhu) [reply cmd/signer/signer.go]: Done.

File: cmd/signer/signer_test.go (hunk 3 above)
Comment (cendhu) [reply cmd/signer/signer_test.go]: Done.

File: cmd/signer/signer_test.go (hunk 4 above)
Comment (cendhu) [reply cmd/signer/signer_test.go]: Done.

File: cmd/signer/signer_test.go (hunk 5 above)
Comment (cendhu) [reply cmd/signer/signer_test.go]: Thanks for finding this. I got confused with how the json marshalled bytes would look like.

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**